crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -27,19 +31,62 @@ use bevy_utils::EntityHashMap;
 /// }
 ///
 /// impl MapEntities for Spring {
-///     fn map_entities(&mut self, entity_mapper: &mut EntityMapper) {
-///         self.a = entity_mapper.get_or_reserve(self.a);
-///         self.b = entity_mapper.get_or_reserve(self.b);
+///     fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M) {
+///         self.a.map_entities(entity_mapper);
+///         self.b.map_entities(entity_mapper)
 ///     }
 /// }
 /// ```
 ///
 pub trait MapEntities {
     /// Updates all [`Entity`] references stored inside using `entity_mapper`.
-    ///
-    /// Implementors should look up any and all [`Entity`] values stored within and
-    /// update them to the mapped values via `entity_mapper`.
-    fn map_entities(&mut self, entity_mapper: &mut EntityMapper);
+    fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M);
+}
+
+/// This traits defines a type that knows how to map [`Entity`] references.
+///
+/// Two implementations are provided:
+/// - `SimpleEntityMapper`: tries to map the [`Entity`] reference, but if it can't, it returns the same [`Entity`] reference.
+/// - `EntityMapper`: tries to map the [`Entity`] reference, but if it can't, it allocates a new [`Entity`] reference.
+pub trait Mapper {
+    /// Map an entity to another entity
+    fn map(&mut self, entity: Entity) -> Entity;
+}
+
+/// Similar to `EntityMapper`, but does not allocate new [`Entity`] references in case we couldn't map the entity.
```

Comment:
The docs here are worse than the explanation above: actually explaining the behavior is better.